### PR TITLE
Fix CI

### PR DESCRIPTION
**Pull request by mat007**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -42,17 +42,16 @@ jobs:
       matrix:
         include:
           # Linux, gcc
-          - { compiler: gcc-5, cxxstd: '14',       boostBranch: boost-1.65.0, os: ubuntu-22.04, container: 'ubuntu:18.04' }
```
Comment (Flamefire) [inline .github/workflows/ci.yml:45]: Only container jobs are affected. There is a workaround to still use the v3 actions which would work but it isn't sure for how long. So I guess we can drop every OS not natively available.

With ubuntu-20.04 we could test GCC 7 so I'd use that as the lowest one. 12 is quite new

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -102,7 +101,9 @@ jobs:
             # Package names are g++-* and clang-*, so set this before correcting CXX for Clang
             pkgs="$CXX xsltproc docbook-xsl docbook-xml lcov ccache"
             sudo apt-get -o Acquire::Retries=$NET_RETRY_COUNT install -y $pkgs
-            CXX="ccache ${CXX/clang-/clang++-}"
+            if [[ "$CXX" =~ clang ]]; then
```
Comment (Flamefire) [inline .github/workflows/ci.yml:104]: Why this change? It disables ccache for GCC

**Review by Flamefire — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mat007) [reply .github/workflows/ci.yml:104]: I’ll try to put it back, but it was failing saying that the gcc binary was not found it path.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -42,17 +42,16 @@ jobs:
...
-          - { compiler: clang-5.0, cxxstd: '14',       boostBranch: boost-1.65.0, os: ubuntu-22.04, container: 'ubuntu:18.04' }
-          - { compiler: clang-5.0, cxxstd: '14',       boostBranch: boost-1.77.0, os: ubuntu-22.04, container: 'ubuntu:18.04' }
-          - { compiler: clang-14,  cxxstd: '14,17,20', boostBranch: master,       os: ubuntu-22.04 }
+          - { compiler: clang-14,  cxxstd: '14,17,20', boostBranch: boost-1.86.0, os: ubuntu-24.04 }
+          - { compiler: clang-18,  cxxstd: '14,17,20', boostBranch: boost-1.86.0, os: ubuntu-24.04 }
+          - { compiler: clang-18,  cxxstd: '14,17,20', boostBranch: master, os: ubuntu-24.04 }
 
-          - { name: Collect coverage, coverage: yes,
-              compiler: gcc-8,     cxxstd: '14',       boostBranch: master,       os: ubuntu-20.04 }
+          - { name: Collect coverage, coverage: yes, compiler: gcc-14, cxxstd: '14', boostBranch: boost-1.86.0, os: ubuntu-24.04 }
```
Comment (Flamefire) [inline .github/workflows/ci.yml:54]: I remember having issues using other versions of GCC/gcov hence I stuck to version 8 here and in other projects

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mat007) [reply .github/workflows/ci.yml:45]: It seems the workaround doesn’t work anymore, see the `main` build failures
```
/__e/node20/bin/node: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.28' not found (required by /__e/node20/bin/node)
```
I.e. https://github.com/actions/checkout/issues/1590

Comment (mat007) [reply .github/workflows/ci.yml:45]: > With ubuntu-20.04 we could test GCC 7 so I'd use that as the lowest one. 12 is quite new

Sure, I’ll do that!

Comment (mat007) [reply .github/workflows/ci.yml:45]: It seems to be failing with gcc 7: https://github.com/mat007/turtle/actions/runs/11085108006/job/30801140786?pr=124

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mat007) [reply .github/workflows/ci.yml]: Right, that was probably something else, I removed this change.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mat007) [reply .github/workflows/ci.yml:45]: Latest release for gcc 7 was 7.5 on November 14, 2019 see https://gcc.gnu.org/releases.html
That looks pretty old to me 🤔

Comment (Flamefire) [reply .github/workflows/ci.yml:45]: GCC 7 just doesn't support C++20 which is that failure. And yes it is 5 yrs old, but for some this is "only 5 yrs". So similar how previously we had GCC 5 with C++14 as the oldest one and we can test GCC 7  easily IMO we should

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -42,17 +42,16 @@ jobs:
       matrix:
         include:
           # Linux, gcc
-          - { compiler: gcc-5, cxxstd: '14',       boostBranch: boost-1.65.0, os: ubuntu-22.04, container: 'ubuntu:18.04' }
-          - { compiler: gcc-5, cxxstd: '14',       boostBranch: boost-1.77.0, os: ubuntu-22.04, container: 'ubuntu:18.04' }
-          - { compiler: gcc-12,cxxstd: '14,17,20', boostBranch: master,       os: ubuntu-22.04 }
-          
+          - { compiler: gcc-8, cxxstd: '14,17,20', boostBranch: boost-1.86.0, os: ubuntu-20.04 }
```
Comment (Flamefire) [inline .github/workflows/ci.yml:45]: How about:
- Oldest: 7
- Oldest with C++20 support: 10 (earlier hat 2a)
- (current) latest: 14 and slightly older 12 with boost-master

- ```suggestion
          - { compiler: gcc-7, cxxstd: '14,17',    boostBranch: boost-1.86.0, os: ubuntu-20.04 }
          - { compiler: gcc-10,cxxstd: '14,17,20', boostBranch: boost-1.86.0, os: ubuntu-22.04 }
          - { compiler: gcc-12,cxxstd: '14,17,20', boostBranch: master,       os: ubuntu-22.04 }
```

Comment (mat007) [reply .github/workflows/ci.yml:45]: Sure!

**Review by Flamefire — APPROVED** [review]
I'd say that works 👍
